src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -176,5 +187,36 @@ internal void AddOrUpdateWorkerChannels(RpcWebHostChannelReadyEvent rpcChannelRe
                         return existingLanguageWorkerChannels;
                     });
         }
+
+        public bool MatchingPlaceholderChannelExists(string selectedRuntime)
+        {
+            // Don't use placeholders if worker runtime not determined for function app
+            if (string.IsNullOrEmpty(selectedRuntime))
+            {
+                return false;
+            }
+
+            // Don't use placeholders if running dotnet
+            if (selectedRuntime.Equals(LanguageWorkerConstants.DotNetLanguageWorkerName, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return false;
+            }
+
+            // Check environment config to see if placeholder worker can be used by the function app
+            if (_placeholderEnvironmentConfig.TryGetValue(selectedRuntime, out Dictionary<string, string> environmentConfig))
+            {
+                foreach (string settingKey in environmentConfig.Keys)
+                {
+                    var currentValue = _environment.GetEnvironmentVariable(settingKey);
```

Comment:
- What about check for runFromZip? 
- As the matching logic is plarform specific i.e on windows look for specific app setting and on linux ignore, split into two methods